libarchive/archive_read_support_format_7zip.c

```diff
@@ -518,6 +519,9 @@ archive_read_format_7zip_bid(struct archive_read *a, int best_bid)
 	if (memcmp(p, _7ZIP_SIGNATURE, 6) == 0)
 		return (48);
 
+	if (memcmp(p, _MSI_SIGNATURE, 8) == 0)
```

Comment:
If you are going to add a negative bid that requires 8 bytes of readahead, you will also need to increase the amount of readahead to 8 on line 514.